### PR TITLE
Add User Guide

### PR DESCRIPTION
**Pull request by csjall**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/niveristand-routing-and-faulting-custom-device/blob/master/CONTRIBUTING.md).

### What does this Pull Request accomplish?

This PR adds a basic user guide.

### Why should this Pull Request be merged?

Documentation can be helpful for understanding how things work.

### What testing has been done?

Reviewed documentation in browser. I am sure there are some typos I have missed. Note the screenshots should be updated once we have finished with the user interface changes.

### CODE REVIEW

File: Docs/User Guide.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
+## Overview
+
+Many HIL test systems require hardware fault injection to simulate faults between the ECU and the rest of the system. The **Routing and Faulting Custom Device** injects faults using routing modules. The **SLSC Switch Custom Device** provides hardware support for SLSC Switch routing modules.
```
Comment (rtzoeller) [inline Docs/User Guide.md]: Should this be "DUT" throughout, instead of "ECU"?

Comment (rtzoeller) [reply Docs/User Guide.md]: I am thinking of the LRU use case.

Comment (jashnani) [reply Docs/User Guide.md]: Agreed. We should replace ECU with DUT....wherever it make sense.

File: Docs/User Guide.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+Each state contains a collection of connections. Each connection consists of a source and destination endpoint on a routing module. By default no connections are added. Add connections to establish connections between the ECU and the I/O. A single connection cannot span multiple routing modules. Define multiple connections, one for each routing module, to connect a signal spanning multiple routing modules.
+
+Bind a ring control on a VeriStand screen to a routing channel to set the current state. Set the current state to change the active connections between the ECU and the I/O. Existing connections from the previous state are disconnected before new connections from the current state are connected. Connections that remain unchanged from the previous state to the current state do not glitch.
+
+![UI Manager](UI%20Manager.png)
+
+## Scripting API
+
+The **Scripting API** use LabVIEW VIs to configure modules, endpoints, aliases, and routing channels in a VeriStand system definition file.
```
Comment (buckd) [inline Docs/User Guide.md]: use -> uses

File: Docs/User Guide.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+The **SLSC Switch Custom Device** enumerates all available endpoints for a routing module. Set the configuration for each endpoint to inform the SLSC Switch driver which endpoints are valid to connect.
+
+| Configuration | Description |
+|---|---|
+| Load | A load endpoint is unrestricted and can be connected to other load and source endpoints. This is the default configuration for an endpoint. |
+| Source | A source endpoint is restricted and cannot be connected to another source endpoint. |
+| Configuration | A configuration endpoint is reserved for internal routing and cannot be used directly. |
+
+```NOTE: Configure the SLSC Switch Custom Device before configuring the Routing and Faulting Custom Device.```
```
Comment (rtzoeller) [inline Docs/User Guide.md]: Incredibly minor, but it bothers me that we're using a code block for emphasis. Is there a better way we can do this, e.g. one of the following?

**NOTE**: Configure the SLSC Switch Custom Device before configuring the Routing and Faulting Custom Device.

---

_NOTE: Configure the SLSC Switch Custom Device before configuring the Routing and Faulting Custom Device._

---

**NOTE**
Configure the SLSC Switch Custom Device before configuring the Routing and Faulting Custom Device.

File: Docs/User Guide.md (hunk 1 above)
Comment (csjall) [reply Docs/User Guide.md]: I replaced all instances of ECU with DUT.

**Comment by csjall** [comment]
> * [ ]  Short to Ground and Short to Power show the same diagram. Is this intentional or is one supposed to short to Fault Bus A?

For now I just jacked the same diagram from some other NI documentation. My plan is to do a 2nd pass and update all the diagrams.

File: Docs/User Guide.md (hunk 3 above)
Comment (csjall) [reply Docs/User Guide.md]: How about?

_**NOTE**: Blah Blah Blah_

That has **bold** and _italics_

File: Docs/User Guide.md [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+The **SLSC Switch Custom Device** enumerates all available endpoints for a routing module. Set the configuration for each endpoint to inform the SLSC Switch driver which endpoints are valid to connect.
+
+| Configuration | Description |
+|---|---|
+| Load | A load endpoint is unrestricted and can be connected to other load and source endpoints. This is the default configuration for an endpoint. |
+| Source | A source endpoint is restricted and cannot be connected to another source endpoint. |
+| Configuration | A configuration endpoint is reserved for internal routing and cannot be used directly. |
+
+_NOTE: Configure the SLSC Switch Custom Device before configuring the Routing and Faulting Custom Device._
+
```
Comment (jashnani) [inline Docs/User Guide.md]: probably worth referencing this white paper in SLSC Switch section or References section: https://www.ni.com/en-us/innovations/white-papers/19/using-the-slsc-12251-2-fault-insertion-and-current-sensing-unit.html

After the note. we could add:
"Here's [some more background](https://www.ni.com/en-us/innovations/white-papers/19/using-the-slsc-12251-2-fault-insertion-and-current-sensing-unit.html) on how to use an SLSC module for fault insertion."

File: Docs/User Guide.md [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+## Routing and Faulting Custom Device
+
+The **Routing and Faulting Custom Device** is a VeriStand add-on that configures aliases and routing channels. By default no routing channels are added. Add routing channels for each collection of connections to be controlled.
+
+Routing Channels are displayed in the VeriStand System Explorer under _Targets >> Controller >> Custom Devices >> Routing and Faulting >> Routing Channels_.
+
+![Routing and Faulting System Explorer](Routing%20and%20Faulting%20System%20Explorer.png)
+
+Each routing channel contains one or more states. Common states for a fault injection application include _Passthrough_, _Open Circuit_, _Short to Ground_, _Short to Power_, and _Pin-to-Pin Short_. By default the _Open Circuit_ state is added. Add additional states to match the requirements of the fault injection application.
+
```
Comment (jashnani) [inline Docs/User Guide.md]: Each routing channel contains one or more states that are user-defined.

**Pull request merged**